salt/crypt.py

```diff
@@ -727,6 +728,10 @@ def sign_in(self, timeout=60, safe=True, tries=1, channel=None):
                             'minion.\nOr restart the Salt Master in open mode to '
                             'clean out the keys. The Salt Minion will now exit.'
                         )
+                        # Add a random sleep here for systems that are using a
+                        # a service manager to immediately restart the service
+                        # to avoid overloading the system
+                        time.sleep(random.randint(10, 20))
```

Comment:
Are you sure the sleep is in the right spot, the next statement is 
`sys.exit(salt.defaults.exitcodes.EX_NOPERM)`    What is the value in sleep then exit.